azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -100,38 +101,39 @@ static void setupWatch(final Map<String, ParseConfigFile> configFileMap) throws
       log.warn("Watchservice was not setup for any config file(s).");
       try {
         watchService.close();
-      } catch (IOException e) {
+      } catch (final IOException e) {
         log.warn("IOException while closing watchService. " + e.getMessage());
       }
       return;
     }
 
-    Runnable runnable = () -> {
+    final Runnable runnable = () -> {
       // Watchservice is established, now listen for the events till eternity!
-      for (;; ) {
-        WatchKey watchKey;
+      for (; ; ) {
+        final WatchKey watchKey;
         try {
           watchKey = watchService.take();
-        } catch (InterruptedException ie) {
+          // Wait for a second to ensure there is only one event for a modification.
```

Comment:
1 second seems like a long time to wait, although it doesn't matter much in this context (file change).

Can you explain more about how having the wait suppresses an event?

Reply:
This question puts some light on it. The author of best answer suggests upto 3 seconds but uses 50 milli seconds in his example code.
https://stackoverflow.com/questions/16777869/java-7-watchservice-ignoring-multiple-occurrences-of-the-same-event

Reply:
Thanks for the link. It  looks like one of the options is checking the last modified time -- you were considering this as well? Can the sleep be shorter, and can you update the comments to say that the 2 ENTRY_MODIFY events would be consolidated into one ENTRY_MODIFY event would have count of 2.